cloud2sql/parquet.py

```diff
@@ -0,0 +1,175 @@
+from resotoclient.models import Kind, Model, Property, JsObject
+from resotolib.types import Json
+from typing import Dict, Tuple, List, Any
+from resotolib.baseresources import BaseResource
+from resotolib import baseresources
+import pyarrow as pa
+import inspect
+from cloud2sql.util import value_in_path
+import json
+
+# This set will hold the names of all "base" resources
+# Since that are abstract classes, there will be no instances of them - hence we do not need a table for them.
+base_kinds = {
+    clazz.kind for _, clazz in inspect.getmembers(baseresources, inspect.isclass) if issubclass(clazz, BaseResource)
+}
+
+
+carz = [
+    Property("cloud", "string"),
+    Property("account", "string"),
+    Property("region", "string"),
+    Property("zone", "string"),
+]
+carz_access = {name: ["ancestors", name, "reported", "id"] for name in ["cloud", "account", "region", "zone"]}
+
+
+class ParquetModel:
+    def __init__(self, model: Model):
+        self.model = model
+        self.table_kinds = [
+            kind
+            for kind in model.kinds.values()
+            if kind.aggregate_root and kind.runtime_kind is None and kind.fqn not in base_kinds
+        ]
+        self.schemas: Dict[str, pa.Schema] = {}
+
+    def table_name(self, kind: str) -> str:
+        replaced = kind.replace(".", "_")
+        return replaced
+
+    def link_table_name(self, from_kind: str, to_kind: str) -> str:
+        # 63 characters to be consistent with SQL model
+        replaced = f"link_{self.table_name(from_kind)[0:25]}_{self.table_name(to_kind)[0:25]}"
+        return replaced
```

Comment:
If we want to be consistent - maybe we should move this logic also to a separate module.